consensus/istanbul/validator/weighted.go

```diff
@@ -359,6 +375,22 @@ func (valSet *weightedCouncil) SubListWithProposer(prevHash common.Hash, propose
 		return []istanbul.Validator{proposer}
 	}
 
+	rules := fork.Rules(new(big.Int).SetUint64(valSet.blockNum + 1))
```

Comment:
Is it always true that `valSet.blockNum + 1 == view.Sequence`?
Can there be a case where `view.Sequence` is Randao but `valStt.blockNum + 1` is not?

Reply:
Fixed to use `fork.Rules(view.Sequence).IsRandao` instead, just like L418 `fork.Rules(view.Sequence).IsIstanbul`.

There is another usage of `fork.Rules(...blockNum+1...)` at `weightedRandomProposer()` selector.
Here, I empirically verified with `TestRandao_Deploy` that `valSet.blockNum + 1` is the currently created block.